streams/runners/storm/common/src/main/java/com/hortonworks/streamline/streams/storm/common/StormRestAPIClient.java

```diff
@@ -45,29 +47,29 @@ public StormRestAPIClient(Client client, String stormApiRootUrl, Subject subject
     }
 
     public Map getTopologySummary(String asUser) {
-        return doGetRequest(getTopologySummaryUrl(asUser));
+        return doGetRequest(getTopologySummaryUrl(encode(asUser)));
     }
 
     public Map getTopology(String topologyId, String asUser) {
-        return doGetRequest(getTopologyUrl(topologyId, asUser));
+        return doGetRequest(getTopologyUrl(encode(topologyId), encode(asUser)));
```

Comment:
nit: May want to push encode into respective methods instead of passing encoded strings to that.
Actually most of these methods generate string of the form like below. This can be extracted to one method which does encoding of topology-id and as-user by passing the given operation.

`stormApiRootUrl + "/topology/" + topologyId + <operation> + <as-user>`

New method can be 
```
    private String generateUrl(String topologyId, String asUser, String operation) {
        String baseUrl = stormApiRootUrl + "/topology/" + encode(topologyId) + "/" + operation;
        if (StringUtils.isNotEmpty(asUser)) {
            baseUrl += "?doAsUser=" + encode(asUser);
        }
        return baseUrl;
    }
```
activation url can be ` generateUrl(topologyId, asUser, "activate")`  which can be used inside activateUrl by removing getTOpologyxxxxUrl methods.